server/src/views/frame.js

```diff
@@ -115,14 +64,14 @@ class Snippet extends React.Component {
     }
 
     return <div className="snippet-container">
-      <a href={'#clip=' + encodeURIComponent(clip.id)}>
+      <a href={'?clip=' + encodeURIComponent(clip.id)}>
         { node }
       </a>
       <p>
         <img ref="commentBubble" className="comment-bubble"
           src={ closed ? this.props.staticLink("img/comment-bubble.png") : this.props.staticLink("img/comment-bubble-open.png") }
           onClick={ this.onClickComment.bind(this) } />
-        <a href={'#clip=' + encodeURIComponent(clip.id)}>
+        <a href={'?clip=' + encodeURIComponent(clip.id)}>
```

Comment:
Yes, it requires the same thing. If it doesn't see the same thing, checksums don't match, it complains, and the entire page is re-rendered on the client instead of just hooking up the client side event handlers and leaving the dom as it was from the server.